packages/query/src/block-processor.ts

```diff
@@ -207,6 +150,9 @@ export class BlockProcessor implements BlockProcessorInterface {
       keepAlive: true,
       abortSignal: this.abortController.signal,
     })) {
+      if (compactBlock.height !== ++currentHeight) {
```

Comment:
The ordering of `++currentHeight` and `currentHeight++` tends to be something that confuses developers. 

Think separate lines would be better:

```ts
currentHeight++;
if (compactBlock.height !== currentHeight) {
  ...
}
```

Reply:
Can we add more details as a code comment above so later developers don't accidentally delete this thinking its unnecessary?

```ts
      // The compact block has a chance not to respect height ordering... (more details)
      // We manual track height to ensure it is valid... (more details)
```